frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -306,13 +307,14 @@ function DatasetEditor(props) {
     if (canBeDataset && isBrandNewDataset) {
       onOpenModal(MODAL_TYPES.SAVE);
     } else if (canBeDataset) {
-      await onSave(dataset);
-      setQueryBuilderMode("view");
+      await onSave(dataset, { rerunQuery: false });
+      await setQueryBuilderMode("view");
+      runQuestionQuery();
```

Comment:
`apiUpdateQuestion` doesn't wait for the query results and we shouldn't do it here, too